### ISSUE TITLE
deallocated_value crashes QEMU when run against nvme drive with metadata

### ISSUE THREAD

**Issue by mikeBashStuff** [spdk/spdk#2649]
Setup is exactly the same as in #2644

```
SPDK:  v22.05-rc1-373-g40d89d602
OS: Fedora35 (5.18.13-100, VM)
QEMU:  7.0.0
```

Running `deallocated_value` against nvme drive, with metadata present, causes qemu instance to crash after being hit with a SIGSEGV on the host side. E.g.:

```
# 0000:00:06.0 is the drive with ms enabled

[root@fedora35-cloud-21845 spdk]# ./test/nvme/deallocated_value/deallocated_value -i 0
TELEMETRY: No legacy callbacks, legacy socket not created
Initializing NVMe Controllers
Attaching to 0000:00:06.0
Attaching to 0000:00:07.0
Attached to 0000:00:06.0
  Namespace ID: 1 size: 6GB
Attached to 0000:00:07.0
  Namespace ID: 1 size: 5GB
Initialization complete.

Controller QEMU NVMe Ctrl       (12341               )
Controller PCI vendor:6966 PCI subsystem vendor:6900
Namespace Block Size:4096
Writing Blocks 0 to 99 with random data.
On next read, read value will match random data.
blocks matching previous data: 100
blocks matching zeroes: 0
blocks matching 0xFF: 0
Deallocating Blocks 0 to 99 with random data.
On next read, read value will match deallocated block read value.
blocks matching previous data: 100
blocks matching zeroes: 0
blocks matching FFh: 0

Controller QEMU NVMe Ctrl       (12340               )
Controller PCI vendor:6966 PCI subsystem vendor:6900
Namespace Block Size:4096
Writing Blocks 0 to 99 with random data.
On next read, read value will match random data.

# QEMU dies, no more output.
```

Backtrace is quite long so leaving it here: 
[qemu_core.zip](https://github.com/spdk/spdk/files/9364851/qemu_core.zip)

**Comment by mikeBashStuff** [comment]
@birkelund I was advised to reach out to you with regards to this issue. Would you kindly take a look? :)

**Comment by jimharris** [comment]
> @birkelund I was advised to reach out to you with regards to this issue. Would you kindly take a look? :)

Basically this SPDK test was not specifying an md buffer when doing I/O to a namespace with non-interleaved metadata.  Not a valid test (which we are addressing) but wanted to bring it to your attention on it causing QEMU to crash.

@mikeBashStuff - this test has really outlived its usefulness.  I don't think updating it to support metadata (interleaved or not) is worth it - so I propose removing the test:  https://review.spdk.io/gerrit/c/spdk/spdk/+/14096.

**Comment by birkelund** [comment]
> > @birkelund I was advised to reach out to you with regards to this issue. Would you kindly take a look? :)
> 

That is a bug and it will be fixed in v7.1.0. You can cherry-pick the fix if you want (upstream commit `51c453266309166c2737623211c0afc12884cccd`).

> Basically this SPDK test was not specifying an md buffer when doing I/O to a namespace with non-interleaved metadata. Not a valid test (which we are addressing) but wanted to bring it to your attention on it causing QEMU to crash.
> 
> @mikeBashStuff - this test has really outlived its usefulness. I don't think updating it to support metadata (interleaved or not) is worth it - so I propose removing the test: https://review.spdk.io/gerrit/c/spdk/spdk/+/14096.

@jimharris, just curious - what assumptions were you validating with that test? I've found that test useful in the past ;)

**Comment by changpe1** [comment]
patch has been merged, so closing this issue for now.
